index.js

```diff
@@ -27,57 +27,70 @@ function checkGuess() {
   const guess = parseInt(guessInput.value, 10);
   attempts = attempts + 1;
 
-  hideAllMessages();
+  const remainingAttempts = maxNumberOfAttempts - attempts;
 
-  if (guess === targetNumber) {
-    numberOfGuessesMessage.style.display = '';
-    numberOfGuessesMessage.innerHTML = `You made ${attempts} guesses`;
+  //this hide function not working. Does it need moved down after creating the function?
+  // I didn't notice a difference when I uncommented it again.
```

Comment:
Depends on the type of function you're calling. 

Example of calling a function before its declared where it is acceptable
```
// Calling the function before its declaration
greet();

// Function declaration
function greet() {
    console.log("Hello, World!");
}

```

Here is an example where calling a function before it is declared is not allowed and will cause errors. 
```
// Calling the function before its declaration does not work
greet();

// Function expression
const greet = function() {
    console.log("Hello, World!");
};

```

Reply:
Also I believe that the` hideAllMessages();` is there  to ensure previous messages are hidden and a clean start for displaying only the current guess's relevant message. So I think we want to leave it there.